app/models/concerns/permalinkable.rb

```diff
@@ -9,7 +9,15 @@ module Permalinkable
 
   module ClassMethods
     def find_by_param!(param)
-      find_by_permalink!(param)
+      begin
+        find_by_permalink!(param)
+      rescue ActiveRecord::RecordNotFound
+        if param =~ /^\d+$/
```

Comment:
Use this to branch at the top of the method, exceptions as flow control isn't good.